File: Baekjoon/Search/수 찾기(1920).py
import sys
input = sys.stdin.readline

N = int(input())
set = set(map(int, input().split()))

M = int(input())
m_list = map(int, input().split())

result = []
for m in m_list:
    result.append('1' if m in set else '0')

print('\n'.join(result))

File: Baekjoon/Search/수 찾기(1920).js
/**
 * N개의 정수 (1~N~100_000)
 * int = 32bit
 * 
 * M개의 정수
 * 
 * if M_INT in N_INT
 */


const rl = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout,
});

const input = [];

rl.on('line', (line) => {
    input.push(line);
});

rl.on('close', () => {
    const set = new Set(input[1].split(' ').map(Number));
    
    const result = [];
    for (const num of input[3].split(' ').map(Number)) {
        result.push(set.has(num) ? 1 : 0);
    }

    console.log(result.join('\n'));
});